# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/mips32/Lowering-mips32.cpp
# Commit: 659b24c3626c
# Full Hash: 659b24c3626c6a299ef34ce4a0109fca2e2a9112
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 21:38:12
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.4: Add tls dependency for WasmTruncateToInt64 and Int64ToFloatingPoint for arm. r=lth
#   
#   In this patch we add a tls dependency for the remaining nodes which use
#   BuiltinThunk to call c++ runtime. By ABI requirements WasmTlsReg should
#   be set.
# ==============================================================================

diff -r 473f1cb675ea -r 659b24c3626c js/src/jit/mips32/Lowering-mips32.cpp
--- a/js/src/jit/mips32/Lowering-mips32.cpp	Fri Sep 11 05:37:42 2020 +0000
+++ b/js/src/jit/mips32/Lowering-mips32.cpp	Fri Sep 11 12:02:42 2020 +0000
@@ -234,6 +234,11 @@
                ins);
 }
 
+void LIRGeneratorMIPS::lowerWasmBuiltinTruncateToInt64(
+    MWasmBuiltinTruncateToInt64* ins) {
+  MOZ_CRASH("We don't use it for this architecture");
+}
+
 void LIRGenerator::visitInt64ToFloatingPoint(MInt64ToFloatingPoint* ins) {
   MDefinition* opd = ins->input();
   MOZ_ASSERT(opd->type() == MIRType::Int64);
@@ -242,3 +247,8 @@
   defineReturn(
       new (alloc()) LInt64ToFloatingPoint(useInt64RegisterAtStart(opd)), ins);
 }
+
+void LIRGeneratorMIPS::lowerBuiltinInt64ToFloatingPoint(
+    MBuiltinInt64ToFloatingPoint* ins) {
+  MOZ_CRASH("We don't use it for this architecture");
+}